pkg/verifier/gcp/entry_point.go

```diff
@@ -23,52 +30,86 @@ const (
-	// if err != nil {
-	// 	return a.Output.AddError(err)
-	// }
+	// Fetch the egress URL list from github, falling back to local lists in the event of a failure.
+	egressListYaml := vei.EgressListYaml
+	var egressListStr, tlsDisabledEgressListStr string
+	if egressListYaml == "" {
+		githubEgressList, githubListErr := egress_lists.GetGithubEgressList(vei.PlatformType)
+		if githubListErr == nil {
+			egressListYaml, githubListErr = githubEgressList.GetContent()
+			if githubListErr == nil {
+				g.Logger.Debug(vei.Ctx, "Using egress URL list from %s at SHA %s", githubEgressList.GetURL(), githubEgressList.GetSHA())
+				egressListStr, tlsDisabledEgressListStr, githubListErr = egress_lists.EgressListToString(egressListYaml, map[string]string{})
+			}
+		}
+		var err error
+		if githubListErr != nil {
+			g.Output.AddError(fmt.Errorf("failed to get egress list from GitHub, falling back to local list: %v", githubListErr))
+			egressListYaml, err = egress_lists.GetLocalEgressList(vei.PlatformType)
+			if err != nil {
+				return g.Output.AddError(err)
+			}
+			egressListStr, tlsDisabledEgressListStr, err = egress_lists.EgressListToString(egressListYaml, map[string]string{})
+			if err != nil {
+				return g.Output.AddError(err)
+			}
+		}
+	}
+
+	// Generate the userData file
+	// Expand replaces all ${var} (using empty string for unknown ones), adding the env variables used in startup-script.sh
+	// Must add fake userDatavariables to replace parts of startup-script.sh that are not env variables but start with $
 	userDataVariables := map[string]string{
-		"AWS_REGION":  "us-east-2", // Not sure if this is the correct data
-		"TIMEOUT":     vei.Timeout.String(),
-		"HTTP_PROXY":  vei.Proxy.HttpProxy,
-		"HTTPS_PROXY": vei.Proxy.HttpsProxy,
-		"CACERT":      base64.StdEncoding.EncodeToString([]byte(vei.Proxy.Cacert)),
-		"NOTLS":       strconv.FormatBool(vei.Proxy.NoTls),
-		"DELAY":       "5",
-		"URLS":        "quay.io",
+		"TIMEOUT":          vei.Timeout.String(),
+		"HTTP_PROXY":       vei.Proxy.HttpProxy,
+		"HTTPS_PROXY":      vei.Proxy.HttpsProxy,
+		"CACERT":           base64.StdEncoding.EncodeToString([]byte(vei.Proxy.Cacert)),
+		"NOTLS":            strconv.FormatBool(vei.Proxy.NoTls),
+		"DELAY":            "5",
+		"URLS":             egressListStr,
+		"TLSDISABLED_URLS": tlsDisabledEgressListStr,
+		"ret":              "${ret}",
+		"?":                "$?",
+		"array[@]":         "${array[@]}",
+		"value":            "$value",
 	}
-	// set probe
-	vei.Probe = dummy.Probe{}
-	userData, err := vei.Probe.GetExpandedUserData(userDataVariables)
+
+	userData, err := vei.Probe.GetExpandedUserData(userDataVariables, startupScript)
 	if err != nil {
 		return g.Output.AddError(err)
 	}
-
 	g.Logger.Debug(vei.Ctx, "Generated userdata script:\n---\n%s\n---", userData)
 
 	if vei.CloudImageID == "" {
-		vei.CloudImageID = cloudImageIDDefault
+		vei.CloudImageID = DEFAULT_CLOUDIMAGEID
```

Comment:
Use `probe.GetMachineImageID()` here instead of a global constant. See also my above comment on that constant